python/cuda_parallel/tests/test_unique_by_key.py

```diff
@@ -0,0 +1,363 @@
+# Copyright (c) 2024-2025, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from typing import List
+
+import cupy as cp
+import numba.cuda
+import numpy as np
+import pytest
+
+import cuda.parallel.experimental.algorithms as algorithms
+import cuda.parallel.experimental.iterators as iterators
+from cuda.parallel.experimental.struct import gpu_struct
+
+DTYPE_LIST = [
+    np.uint8,
+    np.uint16,
+    np.uint32,
+    np.uint64,
+    np.int8,
+    np.int16,
+    np.int32,
+    np.int64,
+    np.float32,
+    np.float64,
+]
+
+
+def random_array(size, dtype, max_value=None) -> np.typing.NDArray:
+    rng = np.random.default_rng()
+    if np.isdtype(dtype, "integral"):
+        if max_value is None:
+            max_value = np.iinfo(dtype).max
+        return rng.integers(max_value, size=size, dtype=dtype)
+    elif np.isdtype(dtype, "real floating"):
+        return rng.random(size=size, dtype=dtype)
+    else:
+        raise ValueError(f"Unsupported dtype {dtype}")
+
+
+def type_to_problem_sizes(dtype) -> List[int]:
+    if dtype in [np.uint8, np.int8]:
+        return [2, 4, 5, 6]
+    elif dtype in [np.uint16, np.int16]:
+        return [4, 8, 14]
+    elif dtype in [np.uint32, np.int32, np.float32]:
+        return [4, 10, 20]
+    elif dtype in [np.uint64, np.int64, np.float64]:
+        return [4, 10, 20]
+    else:
+        raise ValueError("Unsupported dtype")
```

Comment:
This logic is borrowed from reduction algorithm tests, where it is motivated by avoiding overflows in accumulation of positive values for integral types with narrow ranges. 

This algorithm does not have such a limitation, so I would revisit these settings to ensure that maximal arrays stay under reasonable limit, say 0.5Gb. Hence the max problem size of 64-bit types may go as high as 26, for 32-bit types as high as 27,  for 16-bit types to 28, and for 8-bit types up to 29. 

This of course needs to be tapered to keep test run-times reasonable.

But my main point here is that there is no need to keep problem sizes small for narrow integral types.